### DISCUSSION TITLE
MetaMask support for fee currencies

### DISCUSSION THREAD

**Discussion by arthurgousset** [celo-org/rainbowkit-celo#111] [Q&A] [answered]
Copy/pasting a question from @fodisi on [Valora's Discord server](https://discord.com/channels/886972503560433675/887422238071087154/1200650592440291499):

> I saw a MetamaskPlus connector in the celo-rainbowkit repo. I assume we should use this one to offer support for cUSD on Metamask wallets when connected on Celo, correct?

**Comment by arthurgousset** [comment, marked as answer]
Great question @fodisi. 

We worked on an experimental MetaMask snap to investigate how we can add support for fee currencies to MetaMask. 
The experiment was called "gas snap". 

Unfortunately, the MetaMask Snap did not provide the user experience we were hoping for, so we abandonned the "gas snap" experiment for now. 

👉  On that basis, I would NOT recommend you add support for the MetamaskPlus connector in halofi today.

Ref: 
- https://github.com/celo-org/rainbowkit-celo/pull/51
- [metamask-with-fee-currency.ts](https://github.com/celo-org/rainbowkit-celo/blob/12578ee4f06c11cef5f1e51a438cc2611321b369/packages/rainbowkit-celo/wallets/metamask-with-fee-currency.ts#L9)

We should really revert the changes we made and remove the "gas snap" experiment from the rainbowkit config. Thanks for reminding us!

**Comment by fodisi** [comment]
Hi @arthurgousset 
Thanks a lot for the clarification and extra resources 🙏

**Reply by arthurgousset** [reply]
Thanks for asking! Please feel agency to create Github Discussions in any repo you have questions for. That's the best way to get our attention in the cLabs @celo-org/devtooling team.

For example: 
- [celo-org/developer-tooling](https://github.com/celo-org/developer-tooling) (for all things Celo SDKs and CLI)
- [celo-org/faucet](https://github.com/celo-org/faucet) (for all things faucetting)
- [celo-org/staked-celo-web-app/](https://github.com/celo-org/staked-celo-web-app) (for all things stcelo.xyz)

**Comment by arthurgousset** [comment]
I don't have a ton of context on the gas snap @aaronmgdr. Do you think it would be prudent to remove it from the rainbowkit config? Am I missing something?

**Reply by aaronmgdr** [reply]
🪦

**Reply by arthurgousset** [reply]
So effectively reverting part of this PR: 
- https://github.com/celo-org/rainbowkit-celo/pull/51

**Reply by arthurgousset** [reply]
Created an issue to track this: 
- https://github.com/celo-org/rainbowkit-celo/issues/112
